### PR TITLE
Port QuickSwitcher to QML

### PR DESCRIPTION
**Pull request by Jedi18**
QuickSwitcher ported to QML and uses the completer to fetch the rooms.

### CODE REVIEW

**Comment by redsky17** [comment]
The UI looks a lot nicer than it did from your earlier screenshots, so good job =D.  I will say, I'm not sure if the room search algorithm is as good as it could be.  See my example attached where I search for 'water' and the result I want is all the way at the bottom.  I'm not sure if maybe it's a sorting issue or not.  Are you using nico's new Trie implementation for fuzzy search?

Screen Shot 2021-02-25 at 5 43 42 PM

**Comment by deepbluev7** [comment]
I think it uses my algorithm, the problem is, that the list is upside down!

File: resources/qml/QuickSwitcher.qml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    Completer {
+        id: completerPopup
+
+        x: roomTextInput.x - 5
+        y: roomTextInput.y + roomTextInput.height + 5
+        width: parent.width + 10
+        completerName: "room"
+        bottomToTop: true
```
Comment (deepbluev7) [inline resources/qml/QuickSwitcher.qml:56]: I think this should be false.

File: resources/qml/QuickSwitcher.qml [hunk 2]
```diff
@@ -0,0 +1,98 @@
+import QtQuick 2.9
+import QtQuick.Controls 2.3
+import im.nheko 1.0
+
+Popup {
+    id: quickSwitcher
+
+    property int textWidth: 48
```
Comment (deepbluev7) [inline resources/qml/QuickSwitcher.qml:8]: Shouldn't this be textHeight?

File: resources/qml/QuickSwitcher.qml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    Overlay.modal: Rectangle {
+        color: "#aa1E1E1E"
+    }
+
+    MatrixTextField {
+        id: roomTextInput
+
+        anchors.fill: parent
+        font.pixelSize: quickSwitcher.textWidth - 18
```
Comment (deepbluev7) [inline resources/qml/QuickSwitcher.qml:27]: I thin textWidth*0.8 or so makes more sense

**Comment by Jedi18** [comment]
Thanks! Yeah the search results are opposite because bottomToTop should have been false not true. Ok made all the required changes.

**Comment by deepbluev7** [comment]
I feel like the switcher is a bit big and the font is not pixel aligned here!

Screenshot_20210226_194410

You may need to set the flag for that on the ListView in the completer.

**Comment by deepbluev7** [comment]
The font color also does not look ideal here, probably because the background color is wrong:

Screenshot_20210226_194610

**Comment by deepbluev7** [comment]
The room switcher may also need to allow more errors in typing, as I can't find some rooms with it anymore.

**Comment by deepbluev7** [comment]
grafik

I guess I need to debug that.

**Comment by Jedi18** [comment]
Thanks, I'll look into it too

**Comment by deepbluev7** [comment]
Okay, I figured out what caused the white rectangle: It was the background of the Popup!

(I used Gammaray to debug that, which is nice :3)

**Comment by deepbluev7** [comment]
Awesome work! Just cleaned up the small visual issues for you, since sometimes Qml is just annoying, but it really just needed a background: null and a Math.round() here and there! Thank you very much!

**Comment by Jedi18** [comment]
Thank you!! I'm definitely checking out GammaRay, looks useful :D

**Pull request merged**